pkg/channel/consolidated/reconciler/controller/kafkachannel.go

```diff
@@ -251,6 +256,60 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, kc *v1beta1.KafkaChannel
 	return newReconciledNormal(kc.Namespace, kc.Name)
 }
 
+func (r *Reconciler) setupSubscriptionStatusWatcher(ctx context.Context, channel *v1beta1.KafkaChannel) error {
+	var err error
+	groupIDPrefix := fmt.Sprintf("kafka.%s.%s", channel.Namespace, channel.Name)
+
+	m := func(cg string) bool {
+		return strings.HasPrefix(cg, groupIDPrefix)
+	}
+	err = r.kafkaWatcher.Watch(string(channel.ObjectMeta.UID), func() {
+		err := r.markSubscriptionReadiness(ctx, channel, r.kafkaWatcher.List(m))
+		if err != nil {
+			logging.FromContext(ctx).Errorw("error updating subscription readiness", zap.Error(err))
+		}
+	})
+	return err
+}
+
+func (r *Reconciler) markSubscriptionReadiness(ctx context.Context, ch *v1beta1.KafkaChannel, cgs []string) error {
+	after := ch.DeepCopy()
+	after.Status.Subscribers = make([]v1.SubscriberStatus, 0)
+
+	for _, s := range ch.Spec.Subscribers {
+		cg := fmt.Sprintf("kafka.%s.%s.%s", ch.Namespace, ch.Name, s.UID)
+		if Find(cgs, cg) {
+			logging.FromContext(ctx).Debugw("marking subscription", zap.Any("subscription", s))
+			after.Status.Subscribers = append(after.Status.Subscribers, v1.SubscriberStatus{
+				UID:                s.UID,
+				ObservedGeneration: s.Generation,
+				Ready:              corev1.ConditionTrue,
+			})
+		}
+	}
```

Comment:
how does this look when we have a few subscribers?

Reply:
sorry I'm not sure I understand the question. can you elaborate?

Reply:
the rendered printed status

Reply:
f we have 400 subscriptions, they must all show in the after.Status.Subscribers cause that's what define they are ready to receive events